src/Conversion/ONNXToTorch/NN/GemmOp.cpp

```diff
@@ -89,135 +91,108 @@ struct ONNXGemmOpToTorchLowering : public ConversionPattern {
-    Value transposeBVal =
-        (transB != 0) ? rewriter.create<AtenTOp>(loc, transposeBType, bTensor)
-                 : bTensor;
+    Value transposeAVal = (transA != 0) ? rewriter.create<AtenTOp>(
+                                              loc, transposeAType, adaptor.A())
+                                        : adaptor.A();
+    Value transposeBVal = (transB != 0) ? rewriter.create<AtenTOp>(
+                                              loc, transposeBType, adaptor.B())
+                                        : adaptor.B();
 
     // Compute Y = alpha * A' * B' + beta * C
     // Scalar multiplication with alpha(alpha * A')
     // and beta(beta * C) values.
+    FloatAttr alpha = adaptor.alphaAttr();
     Value alphaMulResult = NULL, betaMulResult = NULL;
     if (alpha && alpha.getValueAsDouble() != 1.) {
       Value alpha3v = getFloatValue(alpha, rewriter, loc);
-      alphaMulResult = rewriter.create<AtenMulScalarOp>(loc, transposeAType,
-                                                        transposeAVal, alpha3v);
+      alphaMulResult = rewriter.create<AtenMulScalarOp>(
+          loc, transposeAType, transposeAVal, alpha3v);
     }
 
+    ::mlir::FloatAttr beta = adaptor.betaAttr();
     if (beta && beta.getValueAsDouble() != 1.) {
       Value beta3v = getFloatValue(beta, rewriter, loc);
-      betaMulResult =
-          rewriter.create<AtenMulScalarOp>(loc, cTensor.getType(), cTensor, beta3v);
+      betaMulResult = rewriter.create<AtenMulScalarOp>(
+          loc, adaptor.C().getType(), adaptor.C(), beta3v);
     }
 
     // Bmm Operation ((alpha * A') * B')
     Value bmmValue;
+    Type resultType = getTypeConverter()->convertType(op.getResult().getType());
     if (alphaMulResult)
-      bmmValue = rewriter.create<AtenMmOp>(loc, resultType, alphaMulResult,
-                                            transposeBVal);
+      bmmValue = rewriter.create<AtenMmOp>(
+          loc, resultType, alphaMulResult, transposeBVal);
     else
-      bmmValue = rewriter.create<AtenMmOp>(loc, resultType, transposeAVal,
-                                            transposeBVal);
+      bmmValue = rewriter.create<AtenMmOp>(
+          loc, resultType, transposeAVal, transposeBVal);
 
     // Addition ((alpha * A' * B') + (beta * C))
     Value result;
+    Value iOne = getIntValue(1, rewriter, context, loc);
     if (betaMulResult)
-      result = rewriter.create<AtenAddTensorOp>(loc, resultType, bmmValue,
-                                                betaMulResult, iOne);
+      result = rewriter.create<AtenAddTensorOp>(
+          loc, resultType, bmmValue, betaMulResult, iOne);
     else
-      result = rewriter.create<AtenAddTensorOp>(loc, resultType, bmmValue,
-                                                cTensor, iOne);
+      result = rewriter.create<AtenAddTensorOp>(
+          loc, resultType, bmmValue, adaptor.C(), iOne);
 
     rewriter.replaceOpWithNewOp<torch::TorchConversion::ToBuiltinTensorOp>(
```

Comment:
This unrealized conversion cast op is not needed. You can just `replaceOpWithNewOp` for `AtenAddTensorOp`.